metadata/buckets.go

```diff
@@ -182,6 +185,45 @@ func createBucketIfNotExists(tx *bolt.Tx, keys ...[]byte) (*bolt.Bucket, error)
 	return bkt, nil
 }
 
+func namespacesBucketPath() []byte {
+	return bucketKeyVersion
+}
+
+func getNamespacesBucket(tx *bolt.Tx) *bolt.Bucket {
+	return getBucket(tx, namespacesBucketPath())
+}
+
+// Given a namespace string and a bolt transaction
+// return true if the ns has the shared label in it.
+func hasSharedLabel(tx *bolt.Tx, ns string) bool {
+	labelsBkt := getNamespaceLabelsBucket(tx, ns)
+	if labelsBkt == nil {
+		return false
+	}
+	err := labelsBkt.ForEach(func(k, v []byte) error {
+		if string(k) == sharedNamespaceLabel {
+			return errFoundSharedLabel
+		}
+		return nil
+	})
+	return err == errFoundSharedLabel
```

Comment:
Wondering why you're not taking the same approach as in `getSharableBucket` here (doing so would not need the "magic" `errFoundSharedLabel` error);

```suggestion
	cur := labelsBkt.Cursor()
	for k, _ := cur.First(); k != nil; k, _ = cur.Next() {
		if string(k) == sharedNamespaceLabel {
			return true
		}
	}
	return false
```

Reply:
That's a good suggestion. Will add it to the next update.